### PR TITLE
feat(hetzner): simple cloud-init support

### PR DESCRIPTION
**Pull request by knopki**
`cloud-init` support, similar to the `azure` provider. Just pass cloud-config content as "user data". Then wait `cloud-init` to finish.

### CODE REVIEW

File: yascheduler/clouds/hetzner.py [hunk 1]
```diff
@@ -40,22 +38,17 @@ def create_node(self):
...
-        for _ in range(10):
-            ssh_conn = SSH_Connection(
-                host=ip, user=self.ssh_user, connect_kwargs=self.ssh_custom_key
-            )
-            try: ssh_conn.run('whoami', hide=True)
-            except: time.sleep(5)
-            else: break
+        # wait node up and ready
+        self._run_ssh_cmd_with_backoff(
+            ip, cmd="cloud-init status --wait", max_wait_time=600
```
Comment (blokhin) [inline yascheduler/clouds/hetzner.py:50]: @knopki should we indeed wait for 600 seconds instead of about 55 seconds? :thinking:

Comment (knopki) [reply yascheduler/clouds/hetzner.py]: @blokhin I think it's ok to wait up to 600 seconds with up to 60 retries every 10 seconds. But the problem is that I misplaced arguments.
Fixed. Up to 60 seconds (retries every 5 seconds) for Hetzner and up to 600 seconds (retries every 10 seconds) for slow-as-hell Azure. Practically, SSH connection is successful on first or second try.

**Comment by knopki** [comment]
Rebased on master.

**Review by blokhin — APPROVED** [review]

**Comment by blokhin** [comment]
Aha I see :smile_cat:

**Pull request merged**